demo/csi/nfs/jobs/controller-plugin.nomad.hcl

```diff
@@ -0,0 +1,32 @@
+# Controller plugins create and manage CSI volumes.
+# This one just creates folders within the NFS mount.
+job "controller" {
+  group "controller" {
+    task "controller" {
+      driver = "docker"
+      csi_plugin {
+        id   = "rocketduck-nfs"
+        type = "controller"
+      }
+      config {
+        # thanks rocketDuck for aiming directly at Nomad :)
+        # https://gitlab.com/rocketduck/csi-plugin-nfs
+        image = "registry.gitlab.com/rocketduck/csi-plugin-nfs:0.6.1"
+        args = [
+          "--type=controller",
+          "--endpoint=${CSI_ENDPOINT}", # provided by csi_plugin{}
+          "--node-id=${attr.unique.hostname}",
+          "--nfs-server=${NFS_ADDRESS}:/srv/nfs",
+          "--log-level=DEBUG",
+        ]
+        privileged   = true   # this particular controller mounts NFS in itself
+        network_mode = "host" # allows rpc.statd to work for remote NFS locking
```

Comment:
Oh thank you!  This must be left over from my previous implementation where I specifically encountered an rpc.statd error that this resolved, I think from democratic-csi's node plugin in particular.

Looks like here too I only need it on the node plugin, due to other tasks' mounts hanging when it's restarted as you say.  For the controller, I think it's the only user of its internal NFS mount, so it seems okay with only privileged=true and default network_mode.  Does that sound right to you?

Reply:
Yeah with NFSv4 you don't need a NLM anymore (I think it is handled in the kernel or so?); with NFSv3 you might have to run with `-o nolock` or similar (https://gitlab.com/rocketduck/csi-plugin-nfs/-/issues/9 is similar).

You are right that one most likely won't need `network_mode = "host"` for the controller since it is an internal mount and not "shared". That said I will probably have to double check whether the controller properly unmounts the share or leaves around stray mounts.